core/src/main/java/io/questdb/cutlass/http/processors/JsonQueryProcessor.java

```diff
@@ -567,6 +533,40 @@ private void compileAndExecuteQuery(
         }
     }
 
+    private void doResumeSend(
+            JsonQueryProcessorState state,
+            HttpConnectionContext context
+    ) throws PeerDisconnectedException, PeerIsSlowToReadException, QueryPausedException {
+        LOG.debug().$("resume [fd=").$(context.getFd()).I$();
+
+        final HttpChunkedResponse response = context.getChunkedResponse();
+        while (true) {
+            try {
+                state.resume(response);
+                break;
+            } catch (SqlException | ImplicitCastException e) {
+                sqlError(context.getChunkedResponse(), state, e, configuration.getKeepAliveHeader());
+                break;
```

Comment:
The actual fix is here. The old code kept looping after a `SqlException`/`ImplicitCastException` which led to NPE.

Other non-test changes in this PR are due to auto-reformatting.

Reply:
Should `state.clearFactory()` be called here? 

It's not closed in `testQueryImplicitCastExceptionInWindowFunctionFirstRecord`

![image](https://github.com/user-attachments/assets/dd864d2b-0fed-43e5-9b25-5f486ef30b34)

Reply:
I tried to keep the same flow as what we had before #5445. This means that the factory is returned back to the query cache. But I agree that in case of a SQL or cast exception this is probably silly.

Marked the query as non-cacheable, so that it gets closed instead of returned back to the cache in e614e1f84e1c6fb97c755481c4ec2ca23817ba36

Also, I've spotted that we cache EXPLAIN queries which seems wrong. Fixed that in the same commit.